src/lib/vm/vm.js

```diff
@@ -359,7 +441,10 @@ class VmStack {
       }
     }
 
-    const attributes = {};
+    const attributes = {
+      ...this.vm.state.forwardedRootProps,
+      ref: this.vm.state.forwardedRef,
+    };
```

Comment:
That's not only for the root component, but for every component of the widget.

Reply:
@evgenykuzyakov Good point - I didn't quite realize that until you mentioned it. What do you think about this change? https://github.com/NearSocial/VM/pull/20/commits/7e066658e37d6f70f4ea16e06f8d1ddafb7ec9ce

Say you have the following widget code:

```
const Button = styled.button`
    background: #f00;
`;

return (
  <div>
    <Button type="button" ref="forwardedRef">
      {props.label}: Forwarded
    </Button>
    <Button type="button">{props.label}: Not Forwarded</Button>
    <div>
      <p>Some text here</p>
    </div>
  </div>
);
```

Using the magical value of `ref="forwardedRef"` will attach the ref and forward the props only on that specified element. This will actually help in more complex use cases where you need to choose a deeply nested child that the ref should be forwarded to.

Reply:
This might work